client-app/src/desktop/tabs/features/NewsPanelModel.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2018 Extremely Heavy Industries Inc.
+ */
+
+import {XH, HoistModel} from '@xh/hoist/core';
+import {action, observable, bindable} from '@xh/hoist/mobx';
+import {escapeRegExp, uniq} from 'lodash';
+import {DataViewModel} from "@xh/hoist/desktop/cmp/dataview";
+import {LocalStore} from "@xh/hoist/data";
+import {newsPanelItem} from "./NewsPanelItem";
+import {fmtCompactDate} from "@xh/hoist/format";
+
+@HoistModel
+export class NewsPanelModel {
+
+    SEARCH_FIELDS = ['title', 'text'];
+
+    viewModel = new DataViewModel({
+        store: new LocalStore({
+            fields: ['title', 'source', 'text', 'url', 'imageUrl', 'author', 'published']
+        }),
+        itemFactory: newsPanelItem
+    });
+
+    @observable.ref sourceOptions = [];
+    @observable lastRefresh = null;
+
+    @bindable sourceFilter = null;
+    @observable.ref textFilter = null;
+
+    @action
+    setTextFilter(filter) {
+        this.textFilter = filter;
+    }
+
+    constructor() {
+        this.addReaction({
+            track: () => [this.sourceFilter, this.textFilter, this.lastRefresh],
+            run: () => this.filterData()
+        });
+    }
+
+    @action
```

Comment:
Let's put filterData(),  completeLoad(), and destroy() below an implementation block marker